crates/wasi-provider/src/states/error.rs

```diff
@@ -0,0 +1,38 @@
+use kubelet::state::prelude::*;
+
+use super::crash_loop_backoff::CrashLoopBackoff;
+use super::registered::Registered;
+use crate::PodState;
+
+#[derive(Default, Debug)]
+/// The Pod failed to run.
+// If we manually implement, we can allow for arguments.
+pub struct Error {
+    pub message: String,
+}
+
+#[async_trait::async_trait]
+impl State<PodState> for Error {
+    async fn next(
+        &self,
+        pod_state: &mut PodState,
+        _pod: &Pod,
+    ) -> anyhow::Result<Transition<Box<dyn State<PodState>>, Box<dyn State<PodState>>>> {
+        pod_state.errors += 1;
+        if pod_state.errors > 3 {
```

Comment:
Nit: Magic number?

Reply:
Most definitely a magic number but I think this will go away with the implementation of exponential backoff